src/Servers/Kestrel/Core/src/LocalhostListenOptions.cs

```diff
@@ -68,14 +68,20 @@ internal ListenOptions Clone(IPAddress address)
         var options = new ListenOptions(new IPEndPoint(address, IPEndPoint!.Port))
         {
             KestrelServerOptions = KestrelServerOptions,
-            Protocols = Protocols,
             DisableAltSvcHeader = DisableAltSvcHeader,
             IsTls = IsTls,
             HttpsOptions = HttpsOptions,
             HttpsCallbackOptions = HttpsCallbackOptions,
             EndpointConfig = EndpointConfig
         };
 
+        // Don't propagate Protocols unless it was set explicitly
+        // (otherwise, the clone will flag the default value as set explicitly)
+        if (ProtocolsSetExplicitly)
+        {
+            options.Protocols = Protocols;
+        }
```

Comment:
Would any of the existing tests fail if `Protocols` or `ProtocolsSetExplicitly` wasn't properly set here?

I think we should add a test for this `Clone` method similar to the one we have for `SniOptionsSelector.CloneSslOptions`. It uses reflection to put all the `SslOptions` property names in a list and makes sure we've checked all the properites in the lst when we're done testing the equality of all the properties after cloning.

https://github.com/dotnet/aspnetcore/blob/600eb9aa53c052ec7327e2399744215dbe493a89/src/Servers/Kestrel/Core/test/SniOptionsSelectorTests.cs#L901-L905

As a nit, I think it also might be simpler to just set `_protocols` and everything else directly rather than add conditional logic.

```csharp
        var options = new ListenOptions(new IPEndPoint(address, IPEndPoint!.Port))
        {
            KestrelServerOptions = KestrelServerOptions,
            // Set _protocols instead of Protocols to avoid side-effects
            _protocols = protocols,
            ProtocolsSetExplicitly = ProtocolsSetExplicitly,
// ...
```

Reply:
Re test: Yep, I should add one.
Re nit: I didn't love the idea of making `_protocols` internal, but I'll follow the local convention.